routing/localchans/manager.go

```diff
@@ -87,6 +87,22 @@ func (r *Manager) UpdatePolicy(newSchema routing.ChannelPolicy,
 		// will be used to report invalid channels later on.
 		delete(unprocessedChans, info.ChannelPoint)
 
+		nonZeroInboundFees := newSchema.InboundFee.Base != 0 ||
+			newSchema.InboundFee.Rate != 0
+
+		// Prevent setting inbound fees on private channels, because
```

Comment:
Probably also want to add a protection here to make sure no combination of inbound and outbound fees turns out negative.

Reply:
do you mean we need to check every outbound policy with the inbound and deny the inbound if it would cause an overall negative fee on a channel combination ? So currently it seems possible to allow for negative fees or ? (outbound + inbound < 0)

Reply:
> That is indeed the question. And the same would be needed for setting an outbound policy - make sure that there exists no combination with an inbound fee policy that turns out negative.
> 
> Alternatively we just allow negative fees. Arguably there is value in that too for further traffic shaping.

I thought about it yesterday at the Review Club. I believe there needs to be a restriction ensuring that for every channel, InboundFee + OutboundFee >= 0.
If there are channels where InboundFee + OutboundFee < 0, it could provide an attacker with the opportunity to steal funds from the node by routing back and forth.

However, the condition InboundFee + OutboundFee >= 0 doesn't have to apply to every combination of channels. For individual forwards, negative fees should also be allowed (e.g., when routing from a sink to a source). This could result in a node runner temporarily having fewer sats on the node, but when the distribution of liquidity across the individual channels returns to its original position, he hasn't lost anything. This would be identical to the status quo when node runners rebalance and pay economically sensible fees for it.

Reply:
I agree with @feelancer21 we need to definitely make sure in the grpc command (when setting the policy changes) that we never change a policy of the same channel so that its outbound + inbound <= 0.

Reply:
outbound + inbound = 0 can be allowed imho, only in the case of a strict negative sum, the policy should be rejected.

Reply:
I am not sure if this extra check makes sense, because the in and out fee of a particular channel are never both used at the same time. It's always the in fee of the incoming channel in combination with the out fee of the outgoing channel.

Reply:
That is true, but you could also make money with two forwards (one incoming and one outgoing). But if we don't allow negative total forward fees, the original comment is no longer relevant.